src/helpers/pollUpdates.ts

```diff
@@ -0,0 +1,37 @@
+import { updateAvailable } from '../app/notifications/actions';
+import { Store } from '../app/types';
+import { APP_ENV, RELEASE_ID } from '../config';
+
+export default (store: Store): () => void => {
+  if (APP_ENV !== 'production' || typeof(window) === 'undefined') {
+    return () => undefined;
+  }
+
+  const document = window.document;
+  let interval: ReturnType<typeof setInterval>;
+  const activateInterval = () => interval = setInterval(poll, 60 * 1000);
+  const visibilityListener = () => document.visibilityState === 'visible'
+    ? activateInterval()
+    : clearInterval(interval);
+
+  document.addEventListener('visibilitychange', visibilityListener);
+
+  const cancel = () => {
+    document.removeEventListener('visibilitychange', visibilityListener);
+    clearInterval(interval);
+  };
+
+  async function poll() {
+    const release = await fetch('/rex/release.json')
+      .then((response) => response.json());
```

Comment:
Will/Should this quietly fail if the computer is not connected to the internet? Or if, for some reason the file is not available?

Reply:
yea i'm ashamed to admit that i actually had that in there before but removed it when it was totally impossible to test... i'll try again